bin/dynamic_config/set_hoc_flags.rb

```diff
@@ -35,24 +35,24 @@ def main(argv)
   case mode
     when 'red'
       Gatekeeper.set('postMilestone', value: false)
-      Gatekeeper.set('hoc_activity_writes_disabled', value: true)
       Gatekeeper.set('async_activity_writes', value: true)
       Gatekeeper.set('tracking_pixel_enabled', value: false)
       Gatekeeper.set('puzzle_rating', value: false)
+      DCDO.set('hoc_activity_sample_weight', 0)  # Don't write to hoc_activity.
```

Comment:
I am editing a copy of this file on production-daemon so I can change it to
whatever we want, no deploy or checkin required.  I do agree, it is
worthwhile to get some trickle of data collection even in the red mode.

On Mon, Dec 7, 2015 at 8:03 AM, ashercodeorg <notifications@github.com>
wrote:

> In bin/dynamic_config/set_hoc_flags.rb
> <https://github.com/code-dot-org/code-dot-org/pull/5918#discussion_r46837953>
> :
>
> >        Gatekeeper.set('async_activity_writes', value: true)
> >        Gatekeeper.set('tracking_pixel_enabled', value: false)
> >        Gatekeeper.set('puzzle_rating', value: false)
> > +      DCDO.set('hoc_activity_sample_weight', 0)  # Don't write to hoc_activity.
>
> In retrospect, I think this should be some large number (some small
> percentage) rather than 0 (no percentage). We can certainly handle a small
> number of writes... And it would be a travesty to turn off data collection
> entirely.
>
> Does making this 100 or 500 or even 1000 seem practical? My suggestion
> would be to make this change with another deploy. And if we do turn to red,
> update the DCDO setting manually after flipping to red.
>
> What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/5918/files#r46837953>.
>